### ISSUE TITLE
Configure email/mailers

### ISSUE THREAD

**Issue by summer-cook** [scientist-softserv/webstore#135]
# from meeting notes 1/19
Email notifications
- Emails will be going out with the URL to the marketplace instead of the webstore.  
-Action items create web hooks and those webhook configs include the mailer config. This calls the web service that is ssmail.  
    - Could change mailer to be webstore aware.
     - Or spin up webstore mailer by duplicating ssmail and then changed the way the actions are set up.  (leaning towards this).  Would need to scope if this action items is for a user from a webstore then only retrieve webstore for webhook configs marked webstore. 
    - Or configure 1signal 

*Schedule follow up to continue discussion on this. 

# related prs
- https://github.com/assaydepot/ss-mailer/pull/201
- https://github.com/assaydepot/rx/pull/21706
- ~https://github.com/assaydepot/rx/pull/21719~
- https://github.com/assaydepot/scientist_api_v2/pull/248
- https://github.com/scientist-softserv/webstore/pull/288

**Comment by alishaevn** [comment]
from [today's notes](https://docs.google.com/document/d/1HlAB_EfjxOe9iTx1rONOzNJB3ibjt-o-8bFgDAd7giI/edit#):
- 2 options:
  - Update our existing ssmailer (not very desirable because an external dev can’t do it)
  - Change how webhook configs are looked up. It would require changes to the webhook in the marketplace.
    - Definitely falls on something for prod dev to do
  - In either case there will be fallbacks because the webhooks need to get sent to whatever system needs to access them. This would be good to have support from softserv on
  - Need to check that softserv has access to ssmailer.
- Chris doesn’t think our webhooks will work with 1signal because of additional middleware
- A decision still needs to be made

**Comment by alishaevn** [comment]
chris, rob and I met today for 20 minutes about this. discussion notes can be found [here](https://assaydepot.slack.com/archives/C03FZDALABG/p1675968022255319).

we still didn't come to a resolution, so I placed another meeting on the calendar for next tuesday.

**Comment by summer-cook** [comment]
Currently blocked by #205

**Comment by alishaevn** [comment]
> Currently blocked by https://github.com/scientist-softserv/webstore/issues/205 Flow chart & planning for mailers

this flow chart decision came as the result of the meeting on feb 15 between chris, rob, summer and I. ref: [meeting notes](https://assaydepot.slack.com/archives/C03FZDALABG/p1676500202848259).

**Comment by alishaevn** [comment]
additional[ related tickets](https://assaydepot.slack.com/archives/C03FZDALABG/p1677111767956869) for the prod team.

**Comment by alishaevn** [comment]
the path forward was decided on #205. this ticket is now unblocked.

**Comment by alishaevn** [comment]
pr: https://github.com/assaydepot/ss-mailer/pull/201

**Comment by alishaevn** [comment]
this ticket is blocked by https://github.com/assaydepot/rx/issues/21705

**Comment by alishaevn** [comment]
also related: https://github.com/assaydepot/rx/issues/21718

**Comment by alishaevn** [comment]
ss-mailer#201, the last of the outstanding pr's, was merged yesterday. once all of the related code makes it to api prod, we can test for final verification.

**Comment by alishaevn** [comment]
the api version is still on `2.78.2`, so this work still can't be tested yet.

**Comment by alishaevn** [comment]
the api version is still on `2.78.2`. 

I signed out and back in to the staging site anyway. I then created a new request and sent a proposal from the backoffice. the email I received is below. in clicking the link I was routed to the webstore, but given a "Did Not Connect: Potential Security Issue" error.

the link to my created request is: https://webstore-staging.vercel.app/requests/c98ac1b9-9b91-467a-b5b7-24c9fc65659a
the link that I was redirected to from the email is: https://www.webstore-staging.vercel.app/requests/c98ac1b9-9b91-467a-b5b7-24c9fc65659a

looks like the email added "www" to the url. once I removed the "www" from the url, I was taken to the correct request with an sow.

<details>
<summary> email </summary>

![Image](https://user-images.githubusercontent.com/29032869/232098831-132d9dbb-5722-4ec3-af09-2de50a8cb24c.png)
</details>

<details>
<summary> security issue </summary>


![Image](https://user-images.githubusercontent.com/29032869/232099027-da8d5b0a-b86a-4416-b8ac-dbbdb49df1be.png)
</details>
